### ISSUE TITLE
Galaxy, Nexus & fringe device support

### ISSUE THREAD

**Issue by lukeswitz** [guardianproject/haven#297]
### *Issue Recap: Nexus, Galaxy & Fringe Device Issues*

**Description**: Camera errors and trouble determining facing or angle using Nexus handsets. Caused by its use of facing and rotation to form what would normally be a 0 or 1. Throws error code -38 when starting the MediaRecorder. This bug renders the app useless as stated in previous issues. 

**Reproducing**: Using Haven and attempting to switch cameras when not actively previewing from the `MonitorActivity` would cause the `Camera.open() `call to return a null camera instance. 

**Buggy Workarounds**: Avoid setting the audio source and format in MediaRecorderActivity. Alternatively, encode using h.263 instead of .264. Remove extraneous file separator from the videoFile path if necessary. This will give you a video file but the dependency used to play it back is old. You have to then export it to view, could be a data limit thing. 

**Conclusions**: Setting the `cameraFacing` variable isn’t possible using a device that doesn’t provide a simple 0,1 camera configuration. Once lost, the app will crash when called a second time. For instance, when visiting MonitorView -> Settings -> Camera Sensitivity the parameters are attempting to be set on a null object after the `Camera.open() `call fails due to lack of a provided variable corresponding to the camera of the device. 

- Causes video monitoring to crash when setting the audio output and encoder because media recorder is already active at that time from an outside process using the microphone
- Camera sensitivity in settings will crash if not accessed directly from the MonitorView
- Modification may help some devices and hurt others, further testing required.

**Comment by lukeswitz** [comment]
Update: I took apart offending devices to run diagnostics. What I found required little testing and is also documented in this Reddit thread:
https://www.reddit.com/r/Android/comments/3rjbo8/nexus5x_marshmallow_camera_problem/

I’m working on a stable way to handle the camera using the old API for now. When the camera ID error occurs, it causes nearly half of the recent issues on the board. 

Video recording is also a culprit; implemented as is will fail on Android < 6.0 (or devices without a properly oriented sensor, no external storage or single camera)

Funny, my first opened issue here was to address the rotation issue, and now here we are. Once stability mentioned above is achieved new features shouldn’t cause further problems.
